### ISSUE TITLE
Doubts regarding linking using emar 

### ISSUE THREAD

**Issue by anutosh491** [emscripten-core/emscripten#22810]
So I am trying to get clang-repl working in the browser. 

1) I have llvm (clang, lld) compiled against emcripten and required libraries present in prefix
2) I am trying to compile the [CppInterOp](https://github.com/compiler-research/CppInterOp) library that is a wrapper over clang-repl.

So what I need is a static build for `CppInterOp` (`libclangCppInterOp.a`) that needs to be linked against 
1) some object files (from source files containaing wrappers over clang-repl)
2) static libraries offered by my llvm build (libclanginterpreter.a etc etc)

So basically I am running `emmake` as pasted below and the following is the `link.txt` that we need for the linking step (I've just made some edits like adding the static libraries like clangInterpreter and add the `emnm` command to see the linking ... so once I edit the link.txt I again run make .... see #22811 for more info on this)
```
opt/emsdk/upstream/emscripten/emar qc libclangCppInterOp.a \
    CMakeFiles/clangCppInterOp.dir/CppInterOp.cpp.o \
    CMakeFiles/clangCppInterOp.dir/DynamicLibraryManager.cpp.o \
    CMakeFiles/clangCppInterOp.dir/DynamicLibraryManagerSymbol.cpp.o \
    CMakeFiles/clangCppInterOp.dir/Paths.cpp.o \
    $PREFIX/lib/libclangInterpreter.a \
    $PREFIX/lib/libclangAST.a \
    $PREFIX/lib/libclangBasic.a \
    $PREFIX/lib/libclangFrontend.a \
    $PREFIX/lib/libclangLex.a \
    $PREFIX/lib/libclangSema.a \
    $PREFIX/lib/liblldWasm.a
    
opt/emsdk/upstream/emscripten/emranlib ../libclangCppInterOp.a
opt/emsdk/upstream/emscripten/emnm -C --defined-only ../libclangCppInterOp.a
```
Now as I use `emnm` above to check the link, I see stuff defined out of the object file (`XX.cpp.o`) but I don't see anything like `clang::Interpreter::parse` etc from (`libYY.a`)

Not sure if i am missing something obvious here. Any help is welcome 

If interested the build is being done by this 
```
export LINK_FLAGS="-O1 -sMODULARIZE -sEXPORT_ES6=1 -sASSERTIONS -sALLOW_MEMORY_GROWTH=1 -sINITIAL_MEMORY=128MB -sTOTAL_STACK=32MB -sMAIN_MODULE=1 --preload-file ${EMSCRIPTEN_SYSROOT}/@/"

# Configure step
emcmake cmake -DCMAKE_BUILD_TYPE=Release               \
    -DUSE_CLING=OFF                                     \
    -DUSE_REPL=ON                                       \
    -DCMAKE_PREFIX_PATH=$PREFIX                         \
    -DLLVM_DIR=$PREFIX                                  \
    -DLLD_DIR=$PREFIX                                   \
    -DClang_DIR=$PREFIX                                 \
    -DBUILD_SHARED_LIBS=OFF                             \
    -DCMAKE_INSTALL_PREFIX=$PREFIX                      \
    -DCMAKE_VERBOSE_MAKEFILE=ON                         \
    -DCMAKE_FIND_ROOT_PATH_MODE_PACKAGE=ON              \
    -DEMSCRIPTEN_SYSROOT=${EMSCRIPTEN_SYSROOT}          \
    ../

# Build and install with maximum verbosity

EMCC_CFLAGS="$LINK_FLAGS" emmake make VERBOSE=1
emmake make -j8 install
```

**Comment by anutosh491** [comment]
A check here might be picking up any symbol and testing it out 
```
(xeus-cpp-wasm-host) anutosh491@Anutoshs-MacBook-Air lib % emnm libclangInterpreter.a | grep '__clang_Interpreter_SetValueNoAlloc' 
00005cf4 T __clang_Interpreter_SetValueNoAlloc
(xeus-cpp-wasm-host) anutosh491@Anutoshs-MacBook-Air lib % emnm libclangCppInterOp.a | grep '__clang_Interpreter_SetValueNoAlloc'.      // fails as of now
```

**Comment by anutosh491** [comment]
Taking inspiration from (https://stackoverflow.com/questions/2157629/linking-static-libraries-to-other-static-libraries)

I guess we can use this 
```
# Create initial libclangCppInterOp.a with the main object files
emar qc ../libclangCppInterOp.a CMakeFiles/clangCppInterOp.dir/CppInterOp.cpp.o \
                                            CMakeFiles/clangCppInterOp.dir/DynamicLibraryManager.cpp.o \
                                            CMakeFiles/clangCppInterOp.dir/DynamicLibraryManagerSymbol.cpp.o \
                                            CMakeFiles/clangCppInterOp.dir/Paths.cpp.o
                                            
# Extract all objects from the additional static libraries
emar x $PREFIX/lib/libclangInterpreter.a \
                   $PREFIX/lib/libclangAST.a \
                   $PREFIX/lib/libclangBasic.a \
                   $PREFIX/lib/libclangFrontend.a \
                   $PREFIX/lib/libclangLex.a \
                   $PREFIX/lib/libclangSema.a \
                   $PREFIX/lib/liblldWasm.a

# Append all extracted objects to libclangCppInterOp.a
emar q ../libclangCppInterOp.a *.o

# Clean up extracted object files
rm -f *.o

# Update the archive symbol table
emranlib ../libclangCppInterOp.a
```
This basically helps link clangInterpreter to CppInterOp which means I have some idea on this issue. My main doubt as addressed in (#22811) still remains
